shanoir-ng-datasets/src/main/java/org/shanoir/ng/processing/carmin/schedule/ExecutionStatusMonitor.java

```diff
@@ -0,0 +1,265 @@
+                LOG.error("file exception :", e);
+                e.getMessage();
+            }
+
+        }
+    }
+
+    /**
+     * 
+     * @param in
+     * @param out
+     * @param carminDatasetProcessing
+     */
+    private void decompressTGZ(File in, File out, CarminDatasetProcessing carminDatasetProcessing) {
+        try (TarArchiveInputStream fin = new TarArchiveInputStream(
+                new GzipCompressorInputStream(new FileInputStream(in)))) {
+            TarArchiveEntry entry;
+            while ((entry = fin.getNextTarEntry()) != null) {
+                if (entry.isDirectory()) {
+                    continue;
+                }
+
+                File cacheFolder = new File(out.getAbsolutePath() + File.separator + "cache");
+                if (!cacheFolder.exists()) {
+                    cacheFolder.mkdirs();
+                }
+
+                File currentFile = new File(cacheFolder, entry.getName());
+
+                IOUtils.copy(fin, new FileOutputStream(currentFile));
+
+                // check all nifti formats
+                for (int i = 0; i < listOfNiftiExt.length; i++) {
+                    if (entry.getName().endsWith(listOfNiftiExt[i])) {
+                        createProcessedDataset(currentFile, cacheFolder.getAbsolutePath(),
+                                carminDatasetProcessing);
+                    }
+                }
+
+            }
+        } catch (IOException e) {
+            LOG.error(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * 
+     * @param zipFilePath
+     * @param destDir
+     * @param carminDatasetProcessing
+     */
+    private void createProcessedDataset(File niiftiFile, String destDir,
+            CarminDatasetProcessing carminDatasetProcessing) {
+        File dir = new File(destDir);
+        // create output directory if it doesn't exist
+        if (!dir.exists())
+            dir.mkdirs();
+
+        try {
+
+            ProcessedDatasetImportJob processedDataset = new ProcessedDatasetImportJob();
+            DatasetProcessing datasetProcessing = datasetProcessingService
+                    .findById(carminDatasetProcessing.getId())
+                    .orElseThrow(() -> new NotFoundException("datasetProcessing not found"));
+
+            Study study = studyRepository.findById(datasetProcessing.getStudyId())
+                    .orElseThrow(() -> new NotFoundException("study not found"));
+
+            processedDataset.setDatasetProcessing(datasetProcessing);
+
+            processedDataset.setProcessedDatasetFilePath(niiftiFile.getAbsolutePath());
+            processedDataset.setProcessedDatasetType(ProcessedDatasetType.RECONSTRUCTEDDATASET);
+            processedDataset.setStudyId(datasetProcessing.getStudyId());
+            processedDataset.setStudyName(study.getName());
+
+            // TODO get a subject for the processedDataset, hard coded for the time being
+            processedDataset.setSubjectId(1L);
```

Comment:
Hi Khalil, thank you for your PR. The hard coded subject here does not seem to be a good idea.
With kind regards, Michael

Reply:
You should find the link to the subject via the DatasetProcessing, that is already in the database and linked to the subject?

Reply:
Dear Michael, 

We don't really understand this part, we had an idea to select the subject in the front, as the subject is not part of the datasetprocessing, or maybe we didn't quite understand the link between them

looking forward to your reply, and Thank you always!

Khalil

Reply:
Dear Khalil,
I propose the following: get the processing and its inputDatasets, if their subjectIds are all the same, use it as subjectId for the processed dataset to create, as the information is correct then. If possible leave it empty, when you have multiple input datasets from different subjects, as it makes no sense to link a processed dataset to a subject, when two subjects have been used to process it. Delete the line with subjectName, as it is only part of the Job, but not stored in the dataset created, so we can forget about it. With kind regards and have a nice weekend, Michael

Reply:
last open change